etna/lib/etna/auth.rb

```diff
@@ -64,6 +72,20 @@ def fail_or_redirect(request, msg = 'You are unauthorized')
       return [ 302, { 'Location' => uri.to_s }, [] ]
     end
 
+      # If the application asks for a code of conduct redirect
+      def cc_redirect(request, msg = 'You are unauthorized')
+        return [ 401, { 'Content-Type' => 'text/html' }, [msg] ] unless application.config(:auth_redirect)
```

Comment:
Seems like most of this could be merged with fail_or_redirect, possibly using request.env. I suspect you can also avoid using an exception this way (i.e., set request.env['etna.redirect'] and defer to fail_or_redirect as usual).